src/bundle/Controller/LocationController.php

```diff
@@ -81,4 +99,38 @@ public function updateDefaultSortOrderAction(
 
         return $this->reloadTab('details', $content->id, $location->id);
     }
+
+    /**
+     * Delete locations based on form submit.
+     *
+     * @param FormInterface $form
+     * @param mixed $redirectLocationId
+     *
+     * @return bool
+     *  Whether to reset location or not.
+     */
+    private function deleteLocationsBasedOnFormSubmit(FormInterface $form, $redirectLocationId)
+    {
+        $locationIds = array_keys($form->get('removeLocations')->getData());
+
+        if ($form->get('delete')->isClicked()) {
+            $this->uiLocationService->deleteLocations($locationIds);
+
+            if (in_array($redirectLocationId, $locationIds)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private function addLocationBasedOnFormSubmit(FormInterface $form, Content $content)
+    {
+        if ($form->get('add')->isClicked()) {
```

Comment:
Well, there is clutter, and there is understandability. The code always ends up somewhere :)

Not a blocker for this merge, but I personally think that it would make more sense to test in the action, and process the action that was tested positive. We call the process methods N times while only 1 can be performed at a time, it's honestly a bit weird :)

About versions using it as well, it's not a magical argument: things can be consistently "bad" ;)